### PR TITLE
GCW-3182-Issues with the Charity/Orders page

### PR DESCRIPTION
**Pull request by abulsayyad123**
### Ticket Link: 

https://jira.crossroads.org.hk/browse/GCW-3182

### What does this PR do?

- Added separate endpoint for fetching organisations orders with pagination
- Preload `orders` in `organisation` model for adding `ordersCount` field in `organisation` serializer.
- Added test cases for the same.

Please review this PR

### CODE REVIEW

File: spec/controllers/api/v1/gc_organisations_controller_spec.rb [hunk 1]
```diff
@@ -88,4 +88,30 @@
       expect(parsed_body).to eq(serialized_gc_organisation)
     end
   end
+
+  describe "GET orders for GC organisation" do
```
Comment (patrixr) [inline spec/controllers/api/v1/gc_organisations_controller_spec.rb:92]: Could you add some tests for permissions ? to make sure that only entitled users are allowed to call the endpoint

File: app/controllers/api/v1/gc_organisations_controller.rb [hunk 2]
```diff
@@ -38,11 +51,23 @@ def organisation_name_serializer
       Api::V1::OrganisationNamesSerializer
     end
 
+    def order_serializer
+      Api::V1::OrderShallowSerializer
+    end
+
+    def serialized_orders(orders)
+      ActiveModel::ArraySerializer.new(
```
Comment (bharat619) [inline app/controllers/api/v1/gc_organisations_controller.rb:59]: 👍

File: app/models/organisation.rb [hunk 3]
```diff
@@ -8,6 +8,8 @@ class Organisation < ActiveRecord::Base
   has_many :orders
   has_many :users, through: :organisations_users
 
+  scope :with_eager_load, -> { includes([:orders]) }
```
Comment (bharat619) [inline app/models/organisation.rb:11]: `with_eager_load` seems very generic

can rename to `scope :with_order, -> { includes(:orders) }`

Comment (abulsayyad123) [reply app/models/organisation.rb:11]: Yup.. 👍

File: app/controllers/api/v1/gc_organisations_controller.rb [hunk 4]
```diff
@@ -28,6 +28,19 @@ def names
       find_record_and_render_json(organisation_name_serializer)
     end
 
+    api :GET, '/v1/organisations/:id/organisation_orders', "List all orders associated with organisation"
```
Comment (steveyken) [inline app/controllers/api/v1/gc_organisations_controller.rb:31]: I realise I'm a little late to the party here but normally the API url would be more like this

`/v1/organisations/1/orders`

Comment (abulsayyad123) [reply app/controllers/api/v1/gc_organisations_controller.rb:31]: Sure.. @steveyken i'll this change in separate PR.

**Pull request merged**